core/src/main/java/org/infinispan/container/EntryFactoryImpl.java

```diff
@@ -209,6 +209,18 @@ private CacheEntry getFromContainer(Object key, boolean isOwner, boolean writeOp
       return null;
    }
 
+   private CacheEntry getFromContainerForRead(Object key, boolean isOwner) {
+      InternalCacheEntry ice = container.get(key);
+      if (trace) {
+         log.tracef("Retrieved from container %s", ice);
+      }
+      if (isOwner) {
+         return ice == null ? NullCacheEntry.getInstance() : ice;
+      } else {
+         return ice == null || !ice.isL1Entry() ? null : ice;
```

Comment:
I could have read the code wrongly but this method is only invoked when it ```isOwner==true`` or ``isL1Enabled==true``(the condition is checked in ```wrapEntryForReading()```. In any case, the entry should be valid